### PR TITLE
feat: add protocol spawn feature

### PR DESCRIPTION
**Pull request by driftluo**
In order to allow users to achieve a fully streamed reading and writing experience, and to bypass the problem of the async trait.

I add an interface that is mutually exclusive with the previous callback model. It can get all the information required by the protocol, suitable for async environment

### CODE REVIEW

File: tentacle/examples/use_spawn.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
+use bytes::Bytes;
```
Comment (doitian) [inline tentacle/examples/use_spawn.rs:1]: Suggest to rename this to `simple_using_spawn.rs` and add a line of comment like:

```
/// Implement simple.rs example using `ProtocolSpawn`.

Comment (doitian) [reply tentacle/examples/use_spawn.rs:1]: This protocol is too simple to demonstrate the advantages of the spawn interface in some scenarios. Could you give a more persuasive example?

Comment (driftluo) [reply tentacle/examples/use_spawn.rs:1]: Its biggest advantage is that it reduces a layer of abstraction, allowing users to master the reading end from a distance. At the same time, the difficulty of using the async environment under callback is eliminated. I personally think that these characteristics can be seen from the simple example. For example, the [`select!`](https://docs.rs/futures/0.3.8/futures/macro.select.html) macro cannot be used under callback

**Comment by doitian** [comment]
I think we need time to stabilize the interface. It's better to enable this using a feature and inform users that the interface are unstable and may change in later releases.

**Review by zhangsoledad — APPROVED** [review]

**Review by doitian — APPROVED** [review]

File: tentacle/src/traits.rs [hunk 2]
```diff
@@ -110,6 +113,25 @@ pub trait SessionProtocol {
     }
 }
 
+/// When the negotiation is completed and the agreement is opened, will call the implementation,
+/// allow users to implement the read processing of the protocol by themselves
+///
+/// Implementing this trait means that streaming reading will become possible, and at the same time,
```
Comment (TheWaWaR) [inline tentacle/src/traits.rs:119]:
```suggestion
/// Implementing this trait means that streaming reading directly from the underlying substream will become possible, and at the same time,
```

**Review by TheWaWaR — APPROVED** [review]

**Pull request merged**